codalab/client/local_bundle_client.py

```diff
@@ -928,7 +943,8 @@ def update_worksheet_items(self, worksheet_info, new_items):
         last_item_id = worksheet_info['last_item_id']
         length = len(worksheet_info['items'])
         worksheet = self.model.get_worksheet(worksheet_uuid, fetch_items=False)
-        check_worksheet_has_all_permission(self.model, self._current_user(), worksheet)
+        if not spec_util.is_dashboard(worksheet_info['name']):
```

Comment:
There's no good place to put this in the configuration process right now.
 ./setup.sh runs but doesn't assume there's a DB yet.  This will be
improved once we have OAuth on the bundle service.

In any case, let's have dashboard be created on the fly still.  The
configuration will then be just 'cl work dashboard'.  Also, if someone
upgrading forgets to run the config script, then things won't break with
errors, which is always good.


On Mon, Feb 8, 2016 at 10:33 PM, ChengshuLi <notifications@github.com>
wrote:

> In codalab/client/local_bundle_client.py
> <https://github.com/codalab/codalab-cli/pull/333#discussion_r52271854>:
>
> > @@ -928,7 +943,8 @@ def update_worksheet_items(self, worksheet_info, new_items):
> >          last_item_id = worksheet_info['last_item_id']
> >          length = len(worksheet_info['items'])
> >          worksheet = self.model.get_worksheet(worksheet_uuid, fetch_items=False)
> > -        check_worksheet_has_all_permission(self.model, self._current_user(), worksheet)
> > +        if not spec_util.is_dashboard(worksheet_info['name']):
>
> Huh.. is there a reason we wanna do this rather than making it part of the
> configuration process? I felt like counting on the admin being the first to
> visit the dashboard is not super robust.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/codalab/codalab-cli/pull/333/files#r52271854>.
>